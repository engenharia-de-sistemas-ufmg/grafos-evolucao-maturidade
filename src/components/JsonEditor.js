import React from 'react';
import PropTypes from 'prop-types';
import AceEditor from 'react-ace';

import 'brace/mode/json';
import 'brace/theme/github';

const JsonEditor = ({ graph, apply }) => {
  let innerGraph = { ...graph };
  const onChange = (v) => {
    innerGraph = v;
  };

  const onApply = () => {
    if (typeof innerGraph === 'string') {
      innerGraph = JSON.parse(innerGraph);
    }
    apply(innerGraph);
  };

  return (
    <div>
      <AceEditor
        value={JSON.stringify(graph, null, ' ')}
        mode="json"
        theme="github"
        name="UNIQUE_ID_OF_DIV"
        editorProps={{
          $blockScrolling: true,
          $tabSize: 2,
        }}
        onChange={onChange}
      />
      <button className="btn--primary" type="button" onClick={onApply}>
        Atualizar
      </button>
    </div>
  );
};

JsonEditor.propTypes = {
  graph: PropTypes.object.isRequired,
  apply: PropTypes.func.isRequired,
};

export default JsonEditor;
