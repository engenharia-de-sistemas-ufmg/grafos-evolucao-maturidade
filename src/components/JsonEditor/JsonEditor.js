import React from 'react';
import PropTypes from 'prop-types';
import AceEditor from 'react-ace';

import 'brace/mode/json';
import 'brace/theme/textmate';
import { graphType } from '../../commons/types';

const JsonEditor = ({
  graph, updateGraph,
}) => {
  let innerGraph = { ...graph };
  const onChange = (value) => { innerGraph = value; };
  const onClick = () => {
    if (typeof innerGraph === 'string') {
      innerGraph = JSON.parse(innerGraph);
    }
    updateGraph(innerGraph);
  };

  const buttonUpdate = (
    <button className="btn-update-graph button is-small is-primary" style={{ float: 'right' }} type="button" onClick={onClick}>
      Atualizar
    </button>
  );

  const editor = (
    <AceEditor
      value={JSON.stringify(graph, null, ' ')}
      mode="json"
      theme="textmate"
      name="ACE_EDITOR"
      width="100%"
      tabSize={2}
      onChange={onChange}
    />
  );

  return (
    <div>      
      {editor}
    </div>
  );
};

JsonEditor.propTypes = {
  graph: graphType.isRequired,
  updateGraph: PropTypes.func.isRequired,
};

export default JsonEditor;
