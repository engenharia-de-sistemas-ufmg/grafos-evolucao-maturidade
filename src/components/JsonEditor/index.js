import { connect } from 'react-redux';
import JsonEditor from './JsonEditor';
import { updateGraph } from '../../actions';

const mapStateToProps = (state) => ({
  graph: state.graph,
});

const mapDispatchToProps = (dispatch) => ({
  updateGraph: (graph) => { dispatch(updateGraph(graph)); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(JsonEditor);
