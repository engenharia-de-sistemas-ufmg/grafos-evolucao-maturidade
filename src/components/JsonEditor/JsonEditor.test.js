import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import JsonEditor from './JsonEditor';

describe('JsonEditor', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot', () => {
    const lang = 'pt';
    const showCode = true;
    const graph = { nodes: [], edges: [] };
    const toggleCode = sinon.spy();
    const updateGraph = sinon.spy();
    const component = shallow(
      <JsonEditor
        lang={lang}
        showCode={showCode}
        toggleCode={toggleCode}
        updateGraph={updateGraph}
        graph={graph}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should trigger updateGraph', () => {
    const lang = 'pt';
    const showCode = true;
    const graph = { nodes: [], edges: [] };
    const toggleCode = sinon.spy();
    const updateGraph = sinon.spy();
    const component = shallow(
      <JsonEditor
        lang={lang}
        showCode={showCode}
        toggleCode={toggleCode}
        updateGraph={updateGraph}
        graph={graph}
      />,
    );
    component.find('.btn-update-graph').simulate('click');
    expect(updateGraph.callCount).toBe(1);
  });
});
