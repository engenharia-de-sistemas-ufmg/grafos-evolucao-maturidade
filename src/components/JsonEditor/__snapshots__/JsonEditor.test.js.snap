// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonEditor should match snapshot 1`] = `
<div>
  <button
    className="btn-update-graph button is-small"
    onClick={[Function]}
    style={
      Object {
        "float": "right",
      }
    }
    type="button"
  >
    Atualizar
  </button>
  <ReactAce
    cursorStart={1}
    editorProps={Object {}}
    enableBasicAutocompletion={false}
    enableLiveAutocompletion={false}
    enableSnippets={false}
    focus={false}
    fontSize={12}
    height="500px"
    highlightActiveLine={true}
    maxLines={null}
    minLines={null}
    mode="json"
    name="ACE_EDITOR"
    navigateToFileEnd={true}
    onChange={[Function]}
    onLoad={null}
    onPaste={null}
    onScroll={null}
    placeholder={null}
    readOnly={false}
    scrollMargin={
      Array [
        0,
        0,
        0,
        0,
      ]
    }
    setOptions={Object {}}
    showGutter={true}
    showPrintMargin={true}
    style={Object {}}
    tabSize={2}
    theme="textmate"
    value="{
 \\"nodes\\": [],
 \\"edges\\": []
}"
    width="100%"
    wrapEnabled={false}
  />
</div>
`;
